src/main/java/com/google/devtools/build/lib/buildtool/PostAnalysisQueryBuildTool.java

```diff
@@ -183,8 +183,15 @@ private void doPostAnalysisQuery(
     if (result.isEmpty()) {
       env.getReporter().handle(Event.info("Empty query results"));
     }
+
+    Iterable<T> callbackResults = aggregateResultsCallback.getResult();
+
+    if (postAnalysisQueryEnvironment.shouldOrderResults()) {
+      callbackResults = postAnalysisQueryEnvironment.orderResults(callbackResults);
```

Comment:
Nit: Could you wrap this with the profiler ([example](https://github.com/databricks/bazel/blob/databricks-bazel-4.2.2/src/main/java/com/google/devtools/build/lib/buildtool/BuildTool.java#L198-L198))? It would give us info on how much time is spent in sorting the result.